### PR TITLE
Table processing

### PR DESCRIPTION
**Pull request by bloodbare**
### Description
Describe the proposed changes made in this PR.

### How was this PR tested?
Describe how you tested this PR.

### CODE REVIEW

File: nucliadb_models/nucliadb_models/search.py [hunk 1]
```diff
@@ -751,6 +783,15 @@ class ChatModel(BaseModel):
...
     )
 
+    max_tokens: Optional[int] = Field(
+        default=None, description="Maximum characters to generate"
+    )
+
+    query_context_images: Dict[str, Image] = Field(
+        default={},
+        description="The information retrieval context for the current query",
+    )  # base64.b64encode(image_file.read()).decode('utf-8')
```
Comment (jotare) [inline nucliadb_models/nucliadb_models/search.py:793]: Did you forgot to remove this comment?

**Review by jotare — APPROVED** [review]
I guess it's all right but I feel you should add some tests to validate functionality and simplify future maintenance

File: nucliadb/nucliadb/search/search/chat/prompt.py [hunk 2]
```diff
@@ -365,11 +376,34 @@ async def build(self) -> tuple[PromptContext, PromptContextOrder]:
...
+                ):
+                    page = paragraph.position.page_number
+                    context.images[paragraph.id] = await get_page_image(
+                        self.kbid, paragraph.id, page
+                    )
+                elif paragraph.reference is not None:
+                    image = paragraph.reference
+                    context.images[paragraph.id] = await get_paragraph_image(
+                        self.kbid, paragraph.id, image
+                    )
```
Comment (lferran) [inline nucliadb/nucliadb/search/search/chat/prompt.py:406]: Could it be that the same image is referenced in more than one paragraph? If so, we would be downloading it twice.

Also, images are serially downloaded right now, which can be slow at query time. Should we parallelize it?

File: nucliadb_models/nucliadb_models/search.py [hunk 3]
```diff
@@ -886,22 +947,27 @@ class ChatRequest(BaseModel):
...
+    )
     rag_strategies: list[RagStrategies] = Field(
         default=[],
         title="RAG context building strategies",
         description="Options for tweaking how the context for the LLM model is crafted. `full_resource` will add the full text of the matching resources to the context. `field_extension` will add the text of the matching resource's specified fields to the context. If empty, the default strategy is used.",  # noqa
     )
+    rag_images_strategies: list[RagImagesStrategies] = Field(
+        default=[],
+        title="RAG image context building strategies",
+        description="Options for tweaking how the image based context for the LLM model is crafted. `page_image` will add the full page image of the matching resources to the context. If empty, the default strategy is used with the image of the paragraph.",  # noqa
```
Comment (lferran) [inline nucliadb_models/nucliadb_models/search.py:969]: You should mention the `paragraph_image` strategy too

File: nucliadb/nucliadb/search/search/query.py [hunk 4]
```diff
@@ -372,7 +432,16 @@ async def parse_relation_search(
         autofilters = []
         relations_search = SearchOptions.RELATIONS in self.features
         if relations_search or self.autofilter:
-            detected_entities = await self._get_detected_entities()
+            if self.query_endpoint_enabled:
+                detected_entities = await self._get_detected_entities()
+            else:
+                query_info_result = await self._get_query_information()
```
Comment (lferran) [inline nucliadb/nucliadb/search/search/query.py:438]: aren't you awaiting it twice? one here and one in line 401?

File: nucliadb_models/nucliadb_models/search.py (hunk 3 above)
Comment (bloodbare) [reply nucliadb_models/nucliadb_models/search.py:969]: Its also images of the page

File: nucliadb/nucliadb/search/search/query.py (hunk 4 above)
Comment (bloodbare) [reply nucliadb/nucliadb/search/search/query.py:444]: No its a common strategy to wait multiple times the same task: https://stackoverflow.com/questions/71970942/can-i-await-the-same-task-multiple-times-in-python

**Pull request merged**